### PR TITLE
Update CODEOWNERS file in .github/CODEOWNERS

### PR DESCRIPTION
**Pull request by mukeshjc**
Add CODEOWNERS file to define code ownership rules.

This PR adds a CODEOWNERS file in `.github/CODEOWNERS` to help manage code review assignments and maintain clear ownership of different parts of the codebase.

### CODE REVIEW

**Comment by pierluc-codes** [comment]
Your codeowner is not working

**Comment by mukeshjc** [comment]
> Your codeowner is not working

Hi @pierluc-codes we are waiting on membership approval for this repo. I have sent the corresponding request link to you on Slack. If you could approve that membership, this error would go away for team-ip-compliance.

**Comment by mukeshjc** [comment]
@pierluc-codes We have obtained the necessary permissions and the new addition of default owner made to this CODEOWNERS file is working. We will remediate the other pre-existing faulty groups in follow-up PRs.

**Review by pierluc-codes — APPROVED** [review]

**Pull request merged**
